### PR TITLE
notes on purging expunged resources

### PR DESCRIPTION
**Pull request by shwstppr**
Feature PR: https://github.com/apache/cloudstack/pull/8999

----
📚 Documentation preview 📚: https://cloudstack-documentation--397.org.readthedocs.build/en/397/

### CODE REVIEW

File: source/adminguide/tuning.rst [hunk 1]
```diff
@@ -143,3 +143,76 @@ Article <http://support.citrix.com/article/CTX126531>`_.The article
 refers to XenServer 5.6, but the same information applies to XenServer 6
 
 
+Purging Expunged Resources
+--------------------------
+
+..note::
```
Comment (DaanHoogland) [inline source/adminguide/tuning.rst:149]:
```suggestion
.. note::
```
I think, not sure but the doc show the literal `..note::`

File: source/adminguide/service_offerings.rst [hunk 2]
```diff
@@ -319,6 +319,12 @@ To create a new compute offering:
    -  **Storage Policy**: Name of the storage policy defined at vCenter, this is applicable only for VMware.
       When a specific Zone is selected, one of the storage policies can be selected from the list box.
 
+   -  **Purge Resources**: Whether to cleanup instance and its associated resource from
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:322]:
```suggestion
   -  **Purge Resources**: Whether to cleanup instance and its associated resources from
```

File: source/adminguide/service_offerings.rst [hunk 3]
```diff
@@ -319,6 +319,12 @@ To create a new compute offering:
...
+      database upon expunge. When set to true, the database records for the offering instances and its
```
Comment (sureshanaparti) [inline source/adminguide/service_offerings.rst:323]:
```suggestion
      database upon expunge. When set to true, the database records for the instances and its
```

File: source/adminguide/tuning.rst [hunk 4]
```diff
@@ -143,3 +143,76 @@ Article <http://support.citrix.com/article/CTX126531>`_.The article
...
+expunged.resources.purge.interval                  86400              Interval (in seconds) for the background task to purge the expunged resources.
+expunged.resources.purge.delay                     300                Initial delay (in seconds) to start the background task to purge the expunged resources task.
+expunged.resources.purge.batch.size                50                 Batch size to be used during expunged resources purging.
+expunged.resources.purge.start.time                (empty)            Start time to be used by the background task to purge the expunged resources. Use format yyyy-MM-dd or yyyy-MM-dd HH:mm:ss.
+expunged.resources.purge.keep.past.days            30                 The number of days in the past from the execution time of the background task to purge the expunged resources for which the expunged resources must not be purged. To enable purging expunged resource till the execution of the background task, set the value to zero.
```
Comment (sureshanaparti) [inline source/adminguide/tuning.rst:180]:
```suggestion
expunged.resources.purge.keep.past.days            30                 The number of days in the past from the execution time of the background task to keep the expunged resources for which the expunged resources must not be purged. To enable purging expunged resource till the execution of the background task, set the value to zero.
```

File: source/adminguide/tuning.rst [hunk 5]
```diff
@@ -143,3 +143,76 @@ Article <http://support.citrix.com/article/CTX126531>`_.The article
...
+
+
+Using configuration in offerings
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+..note::
+  Available only for service offerings
+
+_purgeresources_ configuration for offerings can be used to allow immediately
+purging a resource when it is expunged. The configuration can be set to true or
+false _purgeresources_ parameter while creating the correspoding offerings. The
```
Comment (sureshanaparti) [inline source/adminguide/tuning.rst:210]:
```suggestion
false _purgeresources_ parameter while creating the corresponding offerings. The
```

File: source/adminguide/tuning.rst (hunk 4 above)
Comment (shwstppr) [reply source/adminguide/tuning.rst:180]: I guess the complete sentence makes sense as it is,

```
The number of days in the past from the execution time of the background task to purge the expunged resources for which the expunged resources must not be purged
```
If needed, we can rephrase the whole statement

**Review by borisstoyanov — APPROVED** [review]
LGTM

File: source/adminguide/tuning.rst [hunk 6]
```diff
@@ -143,3 +143,76 @@ Article <http://support.citrix.com/article/CTX126531>`_.The article
...
+_purgeresources_ configuration for offerings can be used to allow immediately
+purging a resource when it is expunged. The configuration can be set to true or
+false _purgeresources_ parameter while creating the corresponding offerings. The
+following global setting can be used to control the delay for purging the
+resource after expunge:
+
+================================================   ================   ===================================================================
+Global setting                                     Default values     Description
+================================================   ================   ===================================================================
+expunged.resource.purge.job.delay                  180                Delay (in seconds) to execute the purging of an expunged resource initiated by the configuration in the offering. Minimum value should be 180 seconds and if a lower value is set then the minimum value will be used.
```
Comment (shwstppr) [inline source/adminguide/tuning.rst:217]:
```suggestion
expunged.resource.purge.job.delay                  180                Delay (in seconds) to execute the purging of the DB records of an expunged resource initiated by the configuration in the offering. Minimum value should be 180 seconds and if a lower value is set then the minimum value will be used.
```

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**